src/transactions/feed/queryHelper.ts

```diff
@@ -278,35 +298,44 @@ function anyNetworkHasTransactionsOnCurrentPage(hasTransactionsOnCurrentPage: {
 
 // Queries for transactions feed for any number of networks in parallel,
 // with optional pagination support.
-async function queryTransactionsFeed({
+async function* queryTransactionsFeed({
   address,
   localCurrencyCode,
   params,
+  setActiveRequests,
 }: {
   address: string | null
   localCurrencyCode: string
   params: Array<{
     networkId: NetworkId
     afterCursor?: string
   }>
-}): Promise<{ [key in NetworkId]?: QueryResponse }> {
-  const results = await Promise.all(
-    params.map(({ networkId, afterCursor }) =>
-      queryChainTransactionsFeed({
-        address,
-        localCurrencyCode,
-        networkId,
-        afterCursor,
-      })
-    )
+  setActiveRequests: (updateFunc: (prevState: ActiveRequests) => ActiveRequests) => void
+}): AsyncGenerator<{ [key in NetworkId]?: QueryResponse }> {
+  const promises = params.map(
+    async ({
+      networkId,
+      afterCursor,
+    }): Promise<{ networkId: NetworkId; result: QueryResponse | Error }> => {
+      setActiveRequests((prev) => ({ ...prev, [networkId]: true }))
+      try {
+        const result = await queryChainTransactionsFeed({
+          address,
+          localCurrencyCode,
+          networkId,
+          afterCursor,
+        })
+        return { networkId, result }
+      } finally {
+        setActiveRequests((prev) => ({ ...prev, [networkId]: false }))
+      }
+    }
   )
 
-  return results.reduce((acc, result, index) => {
-    return {
-      ...acc,
-      [params[index].networkId]: result,
-    }
-  }, {})
+  for (const promise of promises) {
+    const { networkId, result } = await promise
+    yield { [networkId]: result }
```

Comment:
If I understand correctly, this would start all the requests in parallel and then serially wait for requests to complete one after the other right? If so, if we have a slow running request at the first one, it would block all subsequent requests right? And if one of them fails, wouldn't all subsequent requests be ignored even if they succeed? Wonder if we can do a non generator based solution where we just invoke some action that can update tx feed for a single n/w

Reply:
All the requests are started in parallel within the map and you are correct that a slow request would block the subsequent requests from being processed; however, they will not be removed from the activeRequests state. During the next polling cycle, the slower request will be skipped, if it's listed as having an active request, and the other networks will be processed accordingly. 

`handleResult` currently updates the transaction feed for a single network; however, it's called from two hooks and has some heavy lifting interacting fetchedResult state. Dispatching it from `queryTransactionsFeed` might be a bit complicated when we have to deal with empty pages and possible refetches.

Reply:
@satish-ravi I think this is now Fixed in #5528 which uses an alternative approach.